generator/ServiceClientGeneratorLib/Generators/Marshallers/JsonRPCExceptionUnmarshaller.tt

```diff
@@ -37,8 +37,35 @@ namespace <#=this.Config.Namespace #>.Model.Internal.MarshallTransformations
         {
             context.Read();
 
+    <#
+            if  (this.Config.ServiceModel.IsAwsQueryCompatible)
+            {
+#>         
+            string errorCode = errorResponse.Code;
+            if (context.ResponseData.IsHeaderPresent("x-amzn-query-error"))
```

Comment:
minor: We usually keep header keys constants in [HeaderKeys.cs](https://github.com/aws/aws-sdk-net/blob/bda6cd2b29916baa95b0cf8f292cdcc4548e1cfc/sdk/src/Core/Amazon.Util/HeaderKeys.cs#L61) I'd add this there close to the other error-related ones.